app/packages/registration-schema/validations.ts

```diff
@@ -96,12 +96,34 @@ export function validateNodeLicense() {
                 type: 'node_license_missing_fields',
                 translationArgs: {
                     missingFields,
+                    numOfFields: missingFieldsList.length,
                 },
             },
         };
     };
 }
 
+export function validateNodeLicenseYear() {
+    return (_: unknown, __: unknown, ___: unknown, changes: any, content: DraftRegistration) => {
+        let validateYearTarget: string = '';
+        if (content.nodeLicense && content.nodeLicense.year) {
+            validateYearTarget = content.nodeLicense.year;
+        }
+        if (changes.nodeLicense && changes.nodeLicense.year) {
+            validateYearTarget = changes.nodeLicense.year;
+        }
```

Comment:
Just wanted to clarify. Is the first condition when the nodeLicense is reset after choosing a new License, and the second is when the `Year` text field is changed?

Reply:
Correct. Priority is given to objects retrieved on the changes since they have the latest updates. Content here refers to the underlying object being validated.